write-fonts/src/tables/layout.rs

```diff
@@ -706,6 +706,23 @@ mod tests {
             .collect();
 
         assert!(builder.prefer_format_1());
+
+        fn iter_class_items(
+            start: u16,
+            end: u16,
+            cls: u16,
+        ) -> impl Iterator<Item = (GlyphId, u16)> {
+            (start..=end).map(move |gid| (GlyphId::new(gid), cls))
+        }
+
+        // 3 ranges of 4 glyphs at 6 bytes a range should be smaller than writing
+        // out the 3 * 4 classes directly
+        let builder: ClassDefBuilder = iter_class_items(5, 8, 3)
+            .chain(iter_class_items(9, 12, 4))
+            .chain(iter_class_items(13, 16, 5))
+            .collect();
+
+        assert!(!builder.prefer_format_1());
```

Comment:
The output when this fails will not be helpful. If it might be useful I suggest printing the builder in all the asserts.